client/landing/stepper/declarative-flow/internals/index.tsx

```diff
@@ -103,6 +104,15 @@ export const FlowRenderer: React.FC< { flow: Flow; steps: readonly StepperStep[]
 		( select ) => ( select( STEPPER_INTERNAL_STORE ) as StepperInternalSelect ).getStepData(),
 		[]
 	);
+	const queryParams = useQuery();
+
+	let customInitialStepSlug;
+	if ( currentStepRoute === '' && flow.getCustomInitialStep ) {
```

Comment:
I don't think you need to worry about `currentStepRoute === ''`, the redirect is inside a switch and will only match if the step is empty or 404.

Reply:
Good call! I simplified this code block in 3800aa0bd7e0e65c30d359a462dae8e73886cdc4 - we now call `flow.getCustomInitialStep()` if it exists, and no longer depend on `currentStepRoute`.